src/test/java/foodwhere/logic/parser/RFindCommandParserTest.java

```diff
@@ -0,0 +1,35 @@
+package foodwhere.logic.parser;
+
+import static foodwhere.logic.parser.CommandParserTestUtil.assertParseFailure;
+import static foodwhere.logic.parser.CommandParserTestUtil.assertParseSuccess;
+
+import java.util.Arrays;
+
+import org.junit.jupiter.api.Test;
+
+import foodwhere.commons.core.Messages;
+import foodwhere.logic.commands.RFindCommand;
+import foodwhere.model.review.NameContainsStallPredicate;
+
+public class RFindCommandParserTest {
+
+    private RFindCommandParser parser = new RFindCommandParser();
+
+    @Test
+    public void parse_emptyArg_throwsParseException() {
+        assertParseFailure(parser, "     ",
+                String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT, RFindCommand.MESSAGE_USAGE));
+    }
+
+    @Test
+    public void parse_validArgs_returnRFindCommand() {
+        // no leading and trailing whitespaces
+        RFindCommand expectedRFindCommand =
+                new RFindCommand(new NameContainsStallPredicate(Arrays.asList("John", "Eatery")));
+        assertParseSuccess(parser, "John Eatery", expectedRFindCommand);
+
+        // multiple whitespaces between keywords
+        assertParseSuccess(parser, " \n John \n \t Eatery  \t", expectedRFindCommand);
+    }
+
```

Comment:
I think trying to parse null or empty string might be a test to try too. However, I would think the behaviour on finding empty string/spaces would be that it matches everything. What do you think?

Reply:
I stand corrected, it probably matches nothing - since the specification is at least one word needs to be matched.